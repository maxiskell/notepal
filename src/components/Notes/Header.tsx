import React from "react";
import { NoteAction } from "../../redux/actions";

interface IProps {
  isEditorOpen: boolean;
  newNote: (event: React.MouseEvent<HTMLButtonElement>) => NoteAction;
}

const Header: React.FC<IProps> = (props) => (
  <nav className="navbar navbar-light bg-light">
    <span className="navbar-brand mr-auto my-0 h1">Notepal</span>

    {!props.isEditorOpen && (
      <button className="btn btn-primary" onClick={props.newNote}>
        New Note
      </button>
    )}
  </nav>
);

export default Header;
