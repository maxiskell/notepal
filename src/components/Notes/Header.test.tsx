import React from "react";
import { render, fireEvent } from "@testing-library/react";

import Header from "./Header";

const mockNewNote = jest.fn();

test("renders properly", () => {
  const { container } = render(
    <Header isEditorOpen={false} newNote={mockNewNote} />
  );

  expect(container).toMatchSnapshot();
});

test("renders app title", () => {
  const { getByText } = render(
    <Header isEditorOpen={false} newNote={mockNewNote} />
  );
  const title = getByText(/Notepal/i);

  expect(title).toBeInTheDocument();
});

test("render new editor button if editor is closed", () => {
  const { getByText } = render(
    <Header isEditorOpen={false} newNote={mockNewNote} />
  );
  const button = getByText(/New Note/i);

  expect(button).toBeInTheDocument();
});

test("not render new editor button if editor is open", () => {
  const { queryAllByText } = render(
    <Header isEditorOpen={true} newNote={mockNewNote} />
  );
  const button = queryAllByText(/New Note/i);

  expect(button).toHaveLength(0);
});

test("new note button calls corresponding function when clicked", () => {
  const { getByText } = render(
    <Header isEditorOpen={false} newNote={mockNewNote} />
  );
  fireEvent.click(getByText(/New Note/i));

  expect(mockNewNote).toHaveBeenCalledTimes(1);
});
