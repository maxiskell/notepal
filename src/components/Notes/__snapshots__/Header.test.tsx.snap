// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders properly 1`] = `
<div>
  <nav
    class="navbar navbar-light bg-light"
  >
    <span
      class="navbar-brand mr-auto my-0 h1"
    >
      Notepal
    </span>
    <button
      class="btn btn-primary"
    >
      New Note
    </button>
  </nav>
</div>
`;
